plugins/woocommerce/tests/php/src/Internal/DataStores/Orders/OrdersTableQueryTests.php

```diff
@@ -235,4 +236,43 @@ public function test_query_filters() {
 		remove_all_filters( 'woocommerce_orders_table_query_clauses' );
 	}
 
+	/**
+	 * @testdox The pre-query escape hook allows replacing the order query.
+	 */
+	public function test_pre_query_escape_hook() {
```

Comment:
can you also add tests for:
1. without pagination params
2. with invalid result (i.e. pagination is expected but not provided)

Reply:
@vedanshujain when querying via `wc_get_orders` with no arguments, the query args will still have defaults set for `limit`, `page`, and `offset` (all to 10). 

Now when the callback returns `null` for `found_orders` and / or `max_num_pages`,  we'll see that even though the limit query args are set, those properties are `null` and we'll fail / ignore the callback. 

What would you say is the best way forward here? Should we make it mandatory for the callback to return `found_orders` and `max_num_pages` in the case of a default limit, or should we try to work around that somehow?

Reply:
if the orders is null, and if limit, page etc are also null/not set, we can safely assume them to be 0 I think

Reply:
Added tests (not yet pushed) but also discussed that we'll fix invalid results using sensible defaults, like setting `$this->found_orders = count( $this->orders );`.